File: src/stackQueue/기능개발.java
package stackQueue;

import java.util.*;

public class 기능개발 {

    public static int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> qe = new LinkedList<>();
        List<Integer> answerList = new ArrayList<>();

        for(int i = 0 ; i < progresses.length ; i++){
            int requiredTime = (100-progresses[i])%speeds[i] == 0 ? (100-progresses[i])/speeds[i] : (100-progresses[i])/speeds[i] + 1 ;
            qe.add(requiredTime);
        }

        while(!qe.isEmpty()){
            int count = 1;
            int day = qe.poll();
            while(!qe.isEmpty() && qe.peek() <= day){
                count++;
                qe.poll();
            }
            answerList.add(count);
        }

        int[] answer = new int[answerList.size()];
        for(int i = 0 ; i < answerList.size(); i++){
            answer[i] = answerList.get(i);
        }


        return answer;
    }

    public static void main(String[] args) {
        int[] progesses = {93, 30, 55};
        int[] speeds = {1, 30, 5};

        for (int i : solution(progesses,speeds))
            System.out.printf(" " + i);
    }
}

File: 스택+큐/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    left = []
    done = 0
    for i in range(len(progresses)):
        left.append(math.ceil((100 - progresses[i]) / speeds[i]))
        
    day = left[0]
    for i in range(len(left)):
        if left[i] <= day:
            done += 1
        else:
            day = left[i]
            answer.append(done)
            done = 1
            
    answer.append(done)
    
    return answer
